edge_prediction_experiment/01_hetionet_recall_dag.py

```diff
@@ -0,0 +1,348 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py:light
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:snorkeling_full_text]
+#     language: python
+#     name: conda-env-snorkeling_full_text-py
+# ---
+
+# # Measure Disease associations Gene Edge Recall
+
+# This notebook is designed to estimate how many hetionet edges can be recalled from this weakly supervised approach.
+# After the discriminator model has been trained, the last step is to predict whether every candidate sentence is mentions a disease associates gene relationship.
+# Following the predictions, the next step is to group each candidate sentence based on each candidate pair and take the max value of each group.
+# This max value represents the probability that an edge should or does exist.
+# Lastly, once each candidate has been scored I use the precision recall curve to estimate how many edges can be recovered from this method.
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+
+import math
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import plotnine as p9
+import plydata as ply
+import plydata.tidy as ply_tdy
+from sklearn.metrics import precision_recall_curve, roc_curve, auc
+from sqlalchemy import create_engine
+
+# +
+# Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmed_central_db"
+
+# Path subject to change for different os
+database_str = (
+    f"postgresql+psycopg2://{username}:{password}@/{dbname}?host=/var/run/postgresql"
+)
+conn = create_engine(database_str)
+# -
+
+hetionet_dag_map_df = pd.read_csv(
```

Comment:
I'm a little confused by some of the headers in your df. Does "hetionet" = 0 mean that there does not exist an edge between the gene and disease pair?

Looking at the first row, I read that to mean that there is not an `A1BG associates hematologic cancer` edge in hetionet but there is 1 sentence that mentions the gene and disease. Not sure about the 8 sentences total.

Reply:
>I'm a little confused by some of the headers in your df. Does "hetionet" = 0 mean that there does not exist an edge between the gene and disease pair?

Yup that means the edge does not exist in hetionet.

>Looking at the first row, I read that to mean that there is not an A1BG associates hematologic cancer edge in hetionet but there is 1 sentence that mentions the gene and disease. Not sure about the 8 sentences total.

You are close. It means that there are 8 sentences that have those two mentions appear together. The 1 just means the entity pair has a sentence that mentions them.